cloudhub-2/modules/ROOT/pages/ch2-networking-guide.adoc

```diff
@@ -0,0 +1,196 @@
+
+The load balancing service forwards traffic to deployed applications.
+
+=== Upstream HTTPS Traffic
+
+By default, the load balancing service forwards traffic deployed application over HTTP. To use HTTPS:
+
+* Enable the *Last-Mile Security* setting in the *Ingress* tab of the xref:ch2-deploy-private-space.adoc#configure-endpoint-path[deployment settings]. 
+* Configure the application to listen on HTTPS by providing a certificate in the xref:ps-config-domains.adoc[TLS context] of your private space.
+
+The following example shows a Mule application configuration that exposes an HTTPS endpoint:
+
+[source,console,linenums]
+----
+<http:listener-config name="HTTP_Listener_Configuration" protocol="HTTPS" host="0.0.0.0" port="${http.port}" doc:name="HTTP Listener Configuration" >
+		<tls:context name="TLS_Context_Custom_Keystore" doc:name="TLS Context">
+			<tls:key-store type="jks" path="server.jks" keyPassword="" password="" alias="" />
+		</tls:context>
+</http:listener-config>
+----
+
+=== Load Balancer Connections
+
+For each request a client makes through the CloudHub 2.0 load balancer `(<app>.<space-id>.region.cloudhub.io)`, the load balancer maintains two connections: one connection between the client and the load balancer, and another connection between the load balancer and the application. For each connection, the load balancer manages a default idle timeout of 300 seconds that is triggered when no data is sent over either connection. If no data is sent or received during this duration, the load balancer closes both connections.
+
+For connections that take longer than 300 seconds to process from either side, handle the processing asynchronously. You can customize the idle timeout value in the *Advanced* tab of your xref:ps-config-advanced.adoc#configure-http-requests-and-read-response-timeout[private space settings] in Runtime Manager, or via the Mule application xref:mule-runtime::global-settings-configuration.adoc[global configurations] of the  xref:mule-runtime::about-mule-configuration.adoc[Mule configuration file].
+
+== DNS Records
+
+The following DNS records are exposed for your CloudHub 2.0 applications:
+
+[cols="2*a"]
+|===
+|`*.<space-id>.<region>.cloudhub.io`|Applications running inside a private space.
+|`<space-id>.<region>.cloudhub.io`|Record pointing to the load balancer. Use this record as the target of a custom domain CNAME.
+|`*.internal-<space-id>.<region>.cloudhub.io`|Applications running inside a private space. The IP addresses for this DNS record are accessible only within your xref:ch2-private-space-about.adoc[private spaces]. They cannot be accessed within xref:ch2-shared-space-about.adoc[shared spaces].
+|`internal-<space-id>.<region>.cloudhub.io`|Internal load balancer. Use this record as the target of a custom domain CNAME. The IP addresses for this DNS record are accessible only within your private spaces. They cannot be accessed within shared spaces.
```

Comment:
The "Use this record as the target of a custom domain CNAME" comment for the last entry (internal load balancer) is wrong. We currently don't support custom domain CNAME for the internal LB. Is that correct?

Reply:
We don't support a vanity domain for private endpoints. But we can still put the internal DOMAIN as a target, right?

Reply:
SME confirmed that it is correct as it is now.

Reply:
Are we planning to have vanity domains for private endpoints released, before we publish this doc?

For the public endpoint scenario:
1. Customer configures example.com
2. They create a CNAME for example.com, which points to the public endpoint for purposes of DNS resolution, to get the IP.
3. When CH2 LB receives the request, it will match the host header example.com against the configure vanity domain (example.com), and it will match.

For the private endpoint scenario:
1. Let say customer sends request to internal.com
2. They have CNAME for internal.com, that points to the private endpoint, and DNS resolution works.
3. But the problem is when CH2 LB receives the request, it will see the host header as internal.com, and it will be unable to match this against any certificate on the CH2 LB, because (currently) we don't support vanity domains for the private endpoint.

Based on my above examples, I don't understand how that statement is valid when we don't support vanity domains for the private endpoint.